src/dmd/expressionsem.d

```diff
@@ -5784,6 +5784,36 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
               assert(a == b, _d_assert_fail!"=="(a, b));
             }()
             */
+
+            /*
+            Stores the result of an operand expression into a temporary
+            if necessary, e.g. if it is an impure fuction call containing side
+            effects as in https://issues.dlang.org/show_bug.cgi?id=20114
+
+            Params:
+                op = an expression which may require a temporary, it will
+                     be replaced with the temporary access accordingly
+
+            Returns: a combined temporary declaration and initialization or
+                     op if a temporary is not required
```

Comment:
The description of param and return value need to be swapped, something like
```
op = an expression which may require a temporary and get replaced by `(auto tmp = op, tmp)`

Returns: `op` or `tmp` for subsequent access to the possibly promoted operand
```

Reply:
Done